neqo-http3/src/client_events.rs

```diff
@@ -31,10 +31,24 @@ pub enum Http3ClientEvent {
     DataReadable { stream_id: u64 },
     /// Peer reset the stream.
     Reset { stream_id: u64, error: AppError },
-    /// Peer has send STOP_SENDING with error code EarlyResponse, other error will post a reset event.
+    /// Peer has send a STOP_SENDING.
     StopSending { stream_id: u64, error: AppError },
-    ///A new push stream
-    NewPushStream { stream_id: u64 },
+    /// A new push promise.
+    PushPromise {
+        push_id: u64,
+        referenced_stream_id: u64,
+        headers: Vec<u8>,
+    },
+    /// A push response headers are ready.
+    PushHeaderReady {
+        push_id: u64,
+        headers: Option<Vec<Header>>,
+        fin: bool,
+    },
+    /// New bytes are available on a push stream for reading.
+    PushDataReadable { push_id: u64 },
```

Comment:
I made it on purpose that a push is only referred by push_id. I will open an issue and do it after this PR lands.
Note: necko will need to remember stream_id besides push_id, I wanted to avoid that.